can/interfaces/tosun.py

```diff
@@ -0,0 +1,266 @@
+import collections
+import time
+import warnings
+from typing import List, Optional, Tuple, Union, Deque, Any
+
+import can
+import can.typechecking
+import ctypes
+from can import CanInitializationError
+from can.bus import LOG
+
+from tosun import TSCanMessage, TSCanFdMessage, TSMasterException, TosunDevice, TSMasterMessageType
```

Comment:
Suggest adding an entry for the optional dependency `tosun` in `extras_require` in `setup.py`.

Reply:
The ZLG-CAN device has been refactored by using rust and upload to pypi.
[https://pypi.org/project/zlgcan-driver-py/](url)